presto-native-execution/src/test/java/com/facebook/presto/nativeworker/SymlinkManifestGeneratorUtils.java

```diff
@@ -0,0 +1,160 @@
+     * @param symlinkManifestDir the directory where the symlink manifest should be created
+     * @throws IOException if an I/O error occurs while creating the directories or writing the manifest
+     */
+    public static void createSymlinkManifest(Path hiveTableLocation, Path symlinkManifestDir) throws IOException
+    {
+        if (Files.notExists(symlinkManifestDir.getParent())) {
+            Files.createDirectory(symlinkManifestDir.getParent());
+        }
+
+        if (Files.notExists(symlinkManifestDir)) {
+            Files.createDirectory(symlinkManifestDir);
+        }
+
+        Path manifestFilePath = symlinkManifestDir.resolve("symlink_manifest");
+        try (BufferedWriter writer = Files.newBufferedWriter(manifestFilePath, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+            List<String> fileList = new ArrayList<>();
+            collectDataFiles(hiveTableLocation, fileList);
+
+            for (String filePath : fileList) {
+                writer.write(filePath);
+                writer.newLine();
+            }
+        }
+    }
+
+    /**
+     * Recursively collects all non-hidden file paths in the source directory.
+     * If a directory is encountered, the method is called recursively to collect files in subdirectories.
+     *
+     * @param sourceDir the directory to start collecting files from
+     * @param fileList the list where the collected file paths are stored
+     * @throws IOException if an I/O error occurs while accessing the directory or files
+     */
+    private static void collectDataFiles(Path sourceDir, List<String> fileList) throws IOException
+    {
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(sourceDir)) {
+            for (Path entry : stream) {
+                if (!Files.isHidden(entry)) {
+                    if (Files.isDirectory(entry)) {
+                        collectDataFiles(entry, fileList);
+                    }
+                    else {
+                        fileList.add(entry.toString());
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Cleans up (deletes) all files and subdirectories inside the specified directory.
+     * This method walks through the directory tree and deletes files in reverse order to ensure
+     * that directories are deleted after their contents.
+     *
+     * @param directory the root directory to clean up
+     * @throws IOException if an I/O error occurs during file deletion
+     */
+    public static void cleanupSymlinkData(Path directory) throws IOException
+    {
+        Files.walk(directory)
+                .sorted(Comparator.reverseOrder())
+                .forEach(path -> {
+                    try {
+                        Files.delete(path);
+                    }
+                    catch (IOException e) {
+                        log.error(e, "Failed to cleanup symlink data at path: " + directory);
```

Comment:
I'm not sure this message is actionable. If no one will read it or do anything about it, consider removing it.